lib/mail/notify/mail_notify_preview_interceptor.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+##
+# This is a ActionMailer interceptor class for previews from the Notifications API
+#
+# See ActionMailer::Base:
+#
+# https://github.com/rails/rails/blob/85c58ffa364414d74ab1f442218959818225d708/actionmailer/lib/action_mailer/base.rb#L398
+
+class MailNotifyPreviewInterceptor
+  ##
+  # ActionMailer call back when a preview is being generated.
+  #
+  # Transforms the content of the passed in message.
+
+  def self.previewing_email(message)
+    new(message).transform!
+  end
```

Comment:
I am not sure I am following and I am by no means an expert in how ActionMailer works! 

I'll try and walk my understanding and see if that helps at all:

The `MailNotifyPreviewInterceptor` [has to implement](https://github.com/rails/rails/blob/6f0d1ad14b92b9f5906e44740fce8b4f1c7075dc/actionmailer/lib/action_mailer/base.rb#L398) the `previewing_email` class method, which for me, means it should not hold instance variables. Instead we initalize a new instance of the class with the object passed in - I assume otherwise we couldn't have the generic method on our Interceptors?

At this point the instantiated `MailNotifyPreviewInterceptor` is like a tiny service object, it's sole job is to transform the message, more usually by changing the existing contents, but for us, to fetch the preview and render it - this feels like a pretty common pattern to me.

In all honesty, I pretty much copied the only `PreviewInterceptor` I could find that did the job! 😄 

https://github.com/rails/rails/blob/v7.1.3.2/actionmailer/lib/action_mailer/inline_preview_interceptor.rb

Reply:
Gotta head off, but essentially I was wondering if `previewing_email` could do everything, since there seemed to be two wrapper methods and one meaningful one, but I guess the structure is needed!

Reply:
I don't think it can, but would be interested to see where you are headed - maybe we could have a look next week?

Reply:
Yeah I'm down to chat next week!